plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/rest/TestIcebergPolarisCatalogCaseInsensitiveMapping.java

```diff
@@ -0,0 +1,204 @@
+final class TestIcebergPolarisCatalogCaseInsensitiveMapping
+        extends AbstractTestQueryFramework
+{
+    private static final SessionContext SESSION_CONTEXT = new SessionContext("dummy", null, null, ImmutableMap.of(), "user");
+    private RESTSessionCatalog icebergCatalog;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        File warehouseLocation = Files.newTemporaryFolder();
+        TestingPolarisCatalog polarisCatalog = closeAfterClass(new TestingPolarisCatalog(warehouseLocation.getPath()));
+
+        Map<String, String> properties = ImmutableMap.<String, String>builder()
+                .put(CatalogProperties.URI, polarisCatalog.restUri() + "/api/catalog")
+                .put(CatalogProperties.WAREHOUSE_LOCATION, TestingPolarisCatalog.WAREHOUSE)
+                .put(CREDENTIAL, polarisCatalog.oauth2Credentials())
+                .put(SCOPE, "PRINCIPAL_ROLE:ALL")
+                .buildOrThrow();
+
+        RESTSessionCatalog icebergCatalogInstance = new RESTSessionCatalog(
+                config -> HTTPClient.builder(config).uri(config.get(CatalogProperties.URI)).build(), null);
+        icebergCatalogInstance.initialize("test_catalog", properties);
+
+        icebergCatalog = icebergCatalogInstance;
+        closeAfterClass(icebergCatalog);
+
+        return IcebergQueryRunner.builder()
+                .setBaseDataDir(Optional.of(warehouseLocation.toPath()))
+                .addIcebergProperty("iceberg.catalog.type", "rest")
+                .addIcebergProperty("iceberg.rest-catalog.uri", polarisCatalog.restUri() + "/api/catalog")
+                .addIcebergProperty("iceberg.rest-catalog.warehouse", TestingPolarisCatalog.WAREHOUSE)
+                .addIcebergProperty("iceberg.rest-catalog.security", "OAUTH2")
+                .addIcebergProperty("iceberg.rest-catalog.oauth2.credential", polarisCatalog.oauth2Credentials())
+                .addIcebergProperty("iceberg.rest-catalog.oauth2.scope", "PRINCIPAL_ROLE:ALL")
+                .addIcebergProperty("iceberg.register-table-procedure.enabled", "true")
+                .addIcebergProperty("iceberg.rest-catalog.case-insensitive-name-matching", "true")
+                .build();
+    }
+
+    @Test
+    void testCaseInSensitiveMatching()
+    {
+        icebergCatalog.createNamespace(SESSION_CONTEXT, Namespace.of("LeVeL1"));
+        icebergCatalog.createNamespace(SESSION_CONTEXT, Namespace.of("LeVeL1", "LeVeL2"));
+        icebergCatalog.createNamespace(SESSION_CONTEXT, Namespace.of("LeVeL1", "LeVeL2", "LeVeL3"));
+
+        List<Namespace> namespaces = icebergCatalog.listNamespaces(SESSION_CONTEXT, Namespace.of("LeVeL1", "LeVeL2"));
+        assertThat(getOnlyElement(namespaces)).isEqualTo(Namespace.of("LeVeL1", "LeVeL2", "LeVeL3"));
+        assertThat(computeActual("SHOW SCHEMAS").getOnlyColumnAsSet()).isEqualTo(Set.of("information_schema", "tpch", "level1", "level1.level2", "level1.level2.level3"));
```

Comment:
I would recommend using methods in `AbstractIterableAssert` when verifying collection's elements. It provides more friendly message than `isEqualTo` in case of assertion failure.

Reply:
Changed